symbolic-symcache/src/new/writer.rs

```diff
@@ -308,6 +313,113 @@ impl SymCacheConverter {
         }
     }
 
+    #[cfg(feature = "il2cpp")]
+    /// Processes a set of [`UsymSymbols`], passing all mapped symbols into the converter.
+    pub fn process_usym(&mut self, usym: &UsymSymbols) -> Result<(), SymCacheError> {
+        // Assume records they are sorted by address; There's a test that guarantees this
+        let mapped_records = usym.records().filter_map(|r| match r {
+            UsymSourceRecord::Unmapped(_) => None,
+            UsymSourceRecord::Mapped(r) => Some(r),
+        });
+
+        let mut curr_id: String = String::default();
+        let mut function_idx = 0;
+        for record in mapped_records {
+            // like process_symbolic_function, skip functions whose address is too large to fit in a
+            // u32
+            if record.address > u32::MAX as u64 {
+                continue;
+            }
+            let address = record.address as u32;
+
+            // Records that belong to the same function will have the same identifier.
+            // Symbols have GUID-like sections to them that might ensure they're unique across
+            // files, but we'll just include the file name and paths to be very safe.
+            let identifier = format!("{}::{}", record.native_file, record.native_symbol);
+            if identifier != curr_id {
+                function_idx = {
+                    let mut function = transform::Function {
+                        name: record.managed_symbol.clone(),
+                        comp_dir: None,
+                    };
+                    for transformer in &self.transformers.0 {
+                        function = transformer.transform_function(function);
+                    }
+
+                    let string_bytes = &mut self.string_bytes;
+                    let strings = &mut self.strings;
+                    let name_offset = Self::insert_string(string_bytes, strings, &function.name);
+
+                    let (fun_idx, _) = self.functions.insert_full(raw::Function {
+                        name_offset,
+                        comp_dir_offset: u32::MAX,
+                        entry_pc: address,
+                        lang: Language::CSharp as u32,
+                    });
+                    fun_idx
+                };
+            }
+
+            let managed_dir = if record.managed_file_info.dir_str().is_empty() {
+                None
+            } else {
+                Some(record.managed_file_info.dir_str())
+            };
```

Comment:
```suggestion
            let managed_dir = match record.managed_file_info.dir_str() {
                s if s == "" => None,
                s => Some(s),
            };
```

Would this work? Seems weird to call the fn twice.

Reply:
You can also match on `""` instead of `s if s == ""`.